### PR TITLE
refactor(web): Replace deprecated substr with substring

### PR DESCRIPTION
**Pull request by ermshiperete**
Also add a comment to `SourcemappedWorker` explaining the difference to `DefaultWorker`.

@keymanapp-test-bot skip

### CODE REVIEW

File: web/src/app/webview/src/debug-main.ts [hunk 1]
```diff
@@ -8,8 +8,8 @@ import { SourcemappedWorker } from '@keymanapp/lexical-model-layer/web'
   * This can only be done during load when the active script will be the
   * last script loaded.  Otherwise the script must be identified by name.
  */
- var scripts = document.getElementsByTagName('script');
- var ss = scripts[scripts.length-1].src;
- var sPath = ss.substr(0,ss.lastIndexOf('/')+1);
+const scripts = document.getElementsByTagName('script');
+const ss = scripts[scripts.length-1].src;
+const sPath = ss.slice(0,ss.lastIndexOf('/'));
```
Comment (mcdurdin) [inline web/src/app/webview/src/debug-main.ts:13]: I think this needs to be:
```suggestion
const sPath = ss.slice(0,ss.lastIndexOf('/')+1);
```

That is:
```
let path='webview/src/debug-main.ts';
path.substr(0, path.lastIndexOf('/')+1) == 'webview/src/'
path.slice(0, path.lastIndexOf('/')+1) == 'webview/src/'
```

File: web/src/app/webview/src/release-main.ts [hunk 2]
```diff
@@ -8,8 +8,8 @@ import { Worker } from '@keymanapp/lexical-model-layer/web'
   * This can only be done during load when the active script will be the
   * last script loaded.  Otherwise the script must be identified by name.
  */
- var scripts = document.getElementsByTagName('script');
- var ss = scripts[scripts.length-1].src;
- var sPath = ss.substr(0,ss.lastIndexOf('/')+1);
+const scripts = document.getElementsByTagName('script');
+const ss = scripts[scripts.length-1].src;
+const sPath = ss.slice(0,ss.lastIndexOf('/'));
```
Comment (mcdurdin) [inline web/src/app/webview/src/release-main.ts:13]:
```suggestion
const sPath = ss.slice(0,ss.lastIndexOf('/')+1);
```

**Review by mcdurdin — CHANGES_REQUESTED** [review]

**Review by jahorton — COMMENTED** [review]
While _definitely_ a nit, I believe use of `substring` over `slice` would be more consistent with similar cases within the repro.  It's more consistent across our code base, as `substring` is already used in numerous places within Web, while `slice` only appears to be used for arrays.

It also better matches the name used for similar functionality on some of our other platforms.
- Java (for Android) also supports `substring`.
- Developer's layout-builder scripts use `substring`.
- macOS code uses methods prefixed with `substring`.

File: web/src/app/webview/src/debug-main.ts (hunk 1 above)
Comment (ermshiperete) [reply web/src/app/webview/src/debug-main.ts:13]: That would be wrong. The second parameter of `slice` is the index of the last element to include, not the number of elements.

Comment (mcdurdin) [reply web/src/app/webview/src/debug-main.ts]: I tried it tho

**Review by mcdurdin — APPROVED** [review]

**Review by jahorton — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.51-alpha](https://keyman.com/downloads/releases/alpha/18.0.51)

**Pull request merged**